utils/src/utils/Tree.scala

```diff
@@ -22,19 +22,29 @@ object Tree {
     * Also returns the paths from the least common ancestor to each node.
     */
   def LCAWithPaths[T](x: T, y: T)(parent: T => T): (T, Seq[T], Seq[T]) = {
+    getLCAWithPaths(x,y)(parent).getOrElse{
```

Comment:
When this gets used in getStageDistance before unrolling, is `parent` definitely the correct function here for the nodes that sit in Ctrl.Node(*, >=0)? Not `master` or looking at the `DefiningBlk` hierarchy?  

Are there any comments anywhere that describe the new hierarchy and methods that are new in this PR?

Reply:
parent should be the correct one to look at here. It now is guaranteed to ignore stages that won't actually exist in the final unrolled graph, and should correctly capture the buffer depth distances. The DefiningBlk hierarchy especially won't work here since blocks, e.g. in Reduce, aren't necessarily distinct stages.

Attempted to add a few more comments in spatial.data.Ctrl